controllers/machineset_controller.go

```diff
@@ -443,7 +443,8 @@ func shouldExcludeMachine(machineSet *clusterv1.MachineSet, machine *clusterv1.M
 		logger.V(4).Info("Machine is not controlled by machineset", "machine", machine.Name)
```

Comment:
Should we actually log here? If we can avoid this log line we can just return the result of the condition we want. We can also tackle this in a separate issue/PR.